components/dot_menu/dot_menu.jsx

```diff
@@ -116,124 +105,126 @@ export default class DotMenu extends Component {
     render() {
         const isSystemMessage = PostUtils.isSystemMessage(this.props.post);
         const isMobile = Utils.isMobile();
+        const canDelete = PostUtils.canDeletePost(this.props.post);
+        const canEdit = PostUtils.canEditPost(this.props.post, this.editDisableAction); // Fix this crazy
 
-        if (this.props.idPrefix === Constants.CENTER && isSystemMessage && !this.state.canDelete && !this.state.canEdit) {
-            return null;
-        }
-
-        if (this.props.idPrefix === Constants.RHS && (this.props.post.state === Constants.POST_FAILED || this.props.post.state === Constants.POST_LOADING)) {
-            return null;
-        }
-
-        let type = 'Post';
-        if (this.props.post.root_id && this.props.post.root_id.length > 0) {
-            type = 'Comment';
-        }
+        const menuItems = [];
 
-        const idPrefix = this.props.idPrefix + 'DotMenu';
-
-        let dotMenuFlag = null;
         if (isMobile && !isSystemMessage) {
-            dotMenuFlag = (
-                <DotMenuFlag
-                    idPrefix={idPrefix + 'Flag'}
-                    idCount={this.props.idCount}
-                    postId={this.props.post.id}
-                    isFlagged={this.props.isFlagged}
-                    actions={{
-                        flagPost: this.props.actions.flagPost,
-                        unflagPost: this.props.actions.unflagPost,
+            let text = (
+                <FormattedMessage id={'rhs_root.mobile.flag'}/>
+            );
+            if (this.props.isFlagged) {
+                text = (
+                    <FormattedMessage id={'rhs_root.mobile.unflag'}/>
+                );
+            }
+            menuItems.push(
+                <DotMenuItem
+                    key={'flag'}
+                    menuItemText={text}
+                    handleMenuItemActivated={() => {
```

Comment:
Can you make all of these callbacks into methods of the DotMenu instead of using a arrow functions so that they work nicer with PureComponent?

Reply:
Fixed